### ISSUE TITLE
Docs are being released on push to main

### ISSUE THREAD

**Issue by FabioRosado** [pyscript/pyscript#937]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a duplication
- [X] I already searched in Google and didn't find any good information or help

### What is the issue/comment/problem?

While helping a user in discord, Jeff [noticed](https://discord.com/channels/972017612454232116/972020206538997822/1040488005850382368) that our docs are being released from main when we update any parts of the documentation.

This caused the user to try the new fetch feature on the latest release of pyscript which wasn’t available yet since we didn’t do a release.

Perhaps we should have a production kind of branch to do releases for both pyscript and the docs?

cc @tedpatrick

**Comment by antocuni** [comment]
Another thing which is semi-related to this is that all code examples in the doc should never use `/latest`, like here for example:
https://github.com/pyscript/pyscript/blob/10a904d8794f0704d8a741375b1eac1542fa244b/docs/tutorials/getting-started.md#L20-L44

Using `/latest` is inherently broken, because it means that all the examples potentially stop to work as soon as we do a new release; and moreover, it encourages people to just copy&paste it, resulting in pyscript app which will break next month.

**Comment by FabioRosado** [comment]
Okay I dug a bit more into this and I see what is happening, we have two GitHub actions:

- One to release `latest` docs
- One to release `stable` docs (upon release)

Looking at the GitHub actions, we seem to copy `redirects.html` which will always send us to `/latest/`, I'll reach out to @jezdez  tomorrow to check if it's okay removing the redirect or if we are doing the redirect for a specific reason 😄

**Comment by jezdez** [comment]
There is also a [third release workflow ](https://github.com/pyscript/pyscript/blob/main/.github/workflows/docs-release.yml) that is creating per-tag docs.

At the time we discussed using GHA to build the docs in three ways:
- [every merge into main would update `/latest/`](https://github.com/pyscript/pyscript/blob/a6280268383d92fe180f32b500a7829aeac7f877/.github/workflows/docs-latest.yml#LL60)
- [every release/tag would create `/<ref-name>/`, e.g. `/1.2.3/`](https://github.com/pyscript/pyscript/blob/a6280268383d92fe180f32b500a7829aeac7f877/.github/workflows/docs-release.yml#LL57)
- [every new PR/branch would create `/review/<pr-number>/`, e.g. `/review/937/`](https://github.com/pyscript/pyscript/blob/a6280268383d92fe180f32b500a7829aeac7f877/.github/workflows/docs-review.yml#LL70)

**Comment by JeffersGlass** [comment]
Thanks @jezdez - for what it's worth Fabio, I believe this usage of `/latest/` corresponds more to how we've been using `/unstable` for release branches.

I know we're on the path toward deprecating `/latest` as a release target, but still, that might be some of the cause of confusion. Users are using `/latest/pyscript.js`, which doesn't match `docs.pyscript.net/latest`, if that makes sense.

**Comment by antocuni** [comment]
yes, I think that the current naming scheme is confusing.
+1 to renaming `docs.pyscript.net/latest` into `/unstable`, so that it matches the URLs that we use to serve `pyscript.js`.

Ideally, I would also like to have `docs.pyscript.net/latest/` to server and/or redirect the latest release, not sure how hard it is to do in practice

**Comment by jezdez** [comment]
I don't have any investment in the naming scheme, the docs infra was after all a hackday project ;) The name "latest" was chosen because of the precedence set by docs hosted on Read The Docs and didn't have anything to do with the hosting of the release files. Obviously, it's your choice how to align with precedence.

If I may suggest, it might be useful to create a `/stable/` folder in [the release workflow](https://github.com/pyscript/pyscript/blob/a6280268383d92fe180f32b500a7829aeac7f877/.github/workflows/docs-release.yml#LL57) in addition to `/<ref-name>/` to point to the docs of the most recent release. I would always clear the `stable` S3 bucket directory first though to not have old rendered files linger there.

**Comment by antocuni** [comment]
> I don't have any investment in the naming scheme, the docs infra was after all a hackday project

sure, no blame on you of course and thanks for having done it :)

> The name "latest" was chosen because of the precedence set by docs hosted on Read The Docs

yes, I agree that follow the read the docs convention would be very reasonable.
The problem is that we made some unfortunate choice w.r.t. naming, so for releases we use `/latest` for "the latest official release" and `/unstable` for "the tip of the `main` branch".

And now there are lots of people who published apps using `pyscript.net/latest/pyscript.js`, so there is a high risk of breaking them. What a mess :man_facepalming: 

Basically, I don't care too much about the exact naming, but I care a lot about being consistent between all the various things which we publish. Another of this is `examples`: currently they are published here:
https://pyscript.net/examples/

ideally, I would like to be able to have published examples for every version of pyscript (including unstable): something like:

https://pyscript.net/examples/     <=== redirect to the latest release, 2022.09.1
https://pyscript.net/examples/2022.09.1/
https://pyscript.net/examples/2022.06.1/
https://pyscript.net/examples/unstable/

And of course, it is important to have consistent names between releases, docs and examples.

**Comment by JeffersGlass** [comment]
The original purpose of this issue has been solved by #977 - any merge that touches `/docs` causes the docs to be rebuilt and pushed to `docs/pyscript.net/unstable. Releases push to `docs.pyscript.net/latest` and `docs/pycsript.net/{version_number}`)

Shall we open another issue with the desire to push versions of examples?

**Comment by FabioRosado** [comment]
Opened up the above issue to track the examples case, closing this issue now 😄
